### PR TITLE
Claimables: Claim Panel UI

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
created the ui for claiming a claimable

will clean it up more / add i18n as i plug in functionality

## Screen recordings / screenshots

https://github.com/user-attachments/assets/ba2f6229-19a6-478e-9454-b1824d6fdb90

## What to test
have not tested on android yet, had to update jdk which busted my builds

### CODE REVIEW

File: src/components/asset-list/RecyclerAssetList2/Claimable.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const { accountAddress, nativeCurrency } = useAccountSettings();
+  const { navigate } = extendedState;
+
+  const { data = [] } = useClaimables(
+    {
+      address: accountAddress,
+      currency: nativeCurrency,
+    },
+    {
+      select: data => data?.filter(claimable => claimable.uniqueId === uniqueId),
```
Comment (greg-schrammel) [inline src/components/asset-list/RecyclerAssetList2/Claimable.tsx:21]: if we are only using one claimable isn't it better to do a `data?.find`?

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@benisgold/claim-ui/4128db484d3d71484fb424535c9b47f63cd98f35.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@benisgold/claim-ui/4128db484d3d71484fb424535c9b47f63cd98f35.ipa) for 4128db484d3d71484fb424535c9b47f63cd98f35

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@benisgold/claim-ui/3cb10a4f595c48cbff5641b5a80dfc29dcfe1ec5.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@benisgold/claim-ui/3cb10a4f595c48cbff5641b5a80dfc29dcfe1ec5.ipa) for 3cb10a4f595c48cbff5641b5a80dfc29dcfe1ec5

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import { View } from 'react-native';
+import { IS_IOS } from '@/env';
+import { ButtonPressAnimation } from '@/components/animations';
+import { RouteProp, useRoute } from '@react-navigation/native';
+import { Claimable } from '@/resources/addys/claimables/types';
+import RainbowCoinIcon from '@/components/coin-icon/RainbowCoinIcon';
+import { useTheme } from '@/theme';
+import { FasterImageView } from '@candlefinance/faster-image';
+import { useClaimables } from '@/resources/addys/claimables/query';
+// import { ContextMenuButton } from '@/components/context-menu';
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:17]: rm

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  const { isDarkMode } = useColorMode();
+  const theme = useTheme();
+
+  // const menuConfig = useMemo(() => {
+  //   return {
+  //     menuItems: {},
+  //   };
+  // }, []);
+
+  // const onShowActionSheet = useCallback(() => {}, []);
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:68]: rm if not needed

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+                {'Claim'}
+              </Text>
+            </TextShadow>
+          </Box>
+        }
+        showBackButton={false}
+      />
+
+      <Box alignItems="center" paddingTop="44px" paddingBottom="24px" gap={42}>
+        {/* <Box alignItems="center" gap={20}> */}
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:93]: rm if not needed

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+                    {chainName}
+                  </Text>
+                  <Text align="center" color="labelSecondary" size="icon 12px" weight="heavy">
+                    􀆏
+                  </Text>
+                </Inline>
+              </Box>
+            </ContextMenuButton>
+          </Inline> */}
+        {/* </Box> */}
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:192]: rm if not needed

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+                </Inline>
+              </Box>
+            </AccentColorProvider>
+          </ButtonPressAnimation>
+          <Inline alignVertical="center" space="2px">
+            <Text align="center" color="labelQuaternary" size="icon 10px" weight="heavy">
+              􀵟
+            </Text>
+            <Text color="labelQuaternary" size="13pt" weight="bold">
+              {`$0.01 to claim on ${chainName}`}
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:225]: is this supposed to be static?

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+        TitleComponent={
+          <Box alignItems="center" flexDirection="row" gap={10} justifyContent="center">
+            <Box
+              as={FasterImageView}
+              source={{ url: claimable.iconUrl }}
+              style={{ height: 20, width: 20, borderRadius: 6, borderWidth: 1, borderColor: 'rgba(0, 0, 0, 0.03)' }}
+            />
+            <TextShadow shadowOpacity={0.3}>
+              <Text align="center" color="label" size="20pt" weight="heavy">
+                {'Claim'}
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:84]:
```suggestion
                Claim
```

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+  // const menuConfig = useMemo(() => {
+  //   return {
+  //     menuItems: {},
+  //   };
+  // }, []);
+
+  // const onShowActionSheet = useCallback(() => {}, []);
+
+  const chainName = ChainNameDisplay[claimable.chainId];
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:70]: should we have a fallback here?

File: src/screens/claimables/ClaimClaimablePanel.tsx [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+import { Claimable } from '@/resources/addys/claimables/types';
+import RainbowCoinIcon from '@/components/coin-icon/RainbowCoinIcon';
+import { useTheme } from '@/theme';
+import { FasterImageView } from '@candlefinance/faster-image';
+import { useClaimables } from '@/resources/addys/claimables/query';
+// import { ContextMenuButton } from '@/components/context-menu';
+
+type RouteParams = {
+  ClaimClaimablePanelParams: { uniqueId: string };
+};
```
Comment (walmat) [inline src/screens/claimables/ClaimClaimablePanel.tsx:21]: can you move these to `RootStackParamList`
https://github.com/rainbow-me/rainbow/blob/dd030ca15cc86207c1f5f78d611d20ed07891f40/src/navigation/types.ts#L19

**Review by walmat — CHANGES_REQUESTED** [review]
some changes but overall looking good

**Pull request closed without merging**